rskj-core/src/main/java/org/ethereum/core/BlockHeader.java

```diff
@@ -94,6 +95,9 @@
     private byte[] bitcoinMergedMiningMerkleProof;
     /* The bitcoin protobuf serialized coinbase tx for merged mining */
     private byte[] bitcoinMergedMiningCoinbaseTransaction;
+
+    private byte[] miningForkDetectionData;
```

Comment:
Why is miningForkDetectionData stored separately if it's already stored inside bitcoinMergedMiningMerkleProof ? 
A lot of code has been added because of this (e.g. new parsing in BlockFactory).
If this is because of a performance issue, then it could be computed a single time and then cached, but not received in the constructor.
Changing BlockFactory has implications in BLOCK_MESSAGE which affects the network protocol. I strongly think this may bring unintended consequences and protocol incompatibilities we are not aware of. Take into account there is no RSKIP that this network packed change.

Reply:
It can not be removed from constructor because it must be provided on block building flow.
I did extract the value from `bitcoinMergedMiningCoinbaseTransaction` 👍